python/res/enkf/row_scaling.py

```diff
@@ -103,3 +111,52 @@ def exp_decay(grid, pos, r0, data_index):
 
         for index in range(target_size):
             self[index] = func(index)
+
+    def assign_vector(self, scaling_vector):
+        """Assign tapering value for all elements via a vector.
+
+        The assign_vector() function will resize the row_scaling vector to the
+        number of elements in the scaling_vector, and assign a value to all
+        elements.
+
+        A typical situation might be to load the scaling data as a EclKW
+        instance from a grdecl formatted file. Before the assign_vector() can
+        be called we must transform to an only active elements representation
+        and use a numpy view:
+
+            # Load scaling vector from grdecl file; typically created with
+            # geomodelling software.
+            with open("scaling.grdecl") as fh:
+                kw_global = EclKW.read_grdecl(fh, "SCALING")
+
+            # Create a ecl_kw copy with only the active elements.
+            kw_active = grid.compressed_kw_copy(kw_global)
+
+            # Create a numpy view and invoke the assign_vector() function
+            row_scaling.assign_vector(kw_active.numpy_view())
```

Comment:
If this is typical usage, then is it API somewhere, or do you have to read this doc in order to use it correctly?

Reply:
I perceive this as typical usage - that is why I have added a quite extensive doc string. But I do not really understand your question: ", then is it API somewhere"?

Reply:
Am being naive here, so bear with me. But if you need many preconditions for something to be used, it's a poor API. But perhaps most users do the steps you mention in this docstring? If that's the case, then it's fine as it is.

If users would _have_ to do these steps, and the only thing they really want to do is call `assign_vector`given a ecl_kw and grdecl file—then this "should be API somewhere". Meaning the code in your docstring should be implemented in libres.

Reply:
OK - the fundamental low level operation which is focused on here is
```Python
def assign_vector(self, numpy_scaling_vector)
     ...
```
you do *not* need a grdecl formatted file or a ecl_kw instance beforehand, you could just as well calculate the numpy_scaling vector by other means - for advanced usage that is probably the most interesting. 

Iff people use alternative software (typically geomodelling) to calculate the entire row_scaling vector in one go I think the approach outlined in the doc string makes sense.

Reply:
Thanks! 👍